### PR TITLE
Wire local room logic

### PR DESCRIPTION
**Pull request by weeman1337**
Towards https://github.com/matrix-org/matrix-react-sdk/pull/8612

Contains parts of the other PR:
Add local room logic in places where it should not change anything yet.

element-web notes: none

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/Avatar-test.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        dmRoomMap = {
+            getUserIdForRoomId: jest.fn(),
+        } as unknown as DMRoomMap;
+        DMRoomMap.setShared(dmRoomMap);
+        roomMember = {
+            getMxcAvatarUrl: jest.fn(),
+        } as unknown as RoomMember;
+    });
+
+    it("null for a null room", () => {
```
Comment (kerryarchibald) [inline test/Avatar-test.ts:60]: nitpick: these should follow the `it('should return null....` pattern

File: test/components/views/messages/EncryptionEvent-test.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
```
Comment (kerryarchibald) [inline test/components/views/messages/EncryptionEvent-test.tsx:2]: No worries for these ones, but we should be writing all new unit tests with react testing library

**Review by kerryarchibald — APPROVED** [review]
Left a small nitpick, otherwise LGTM. Thanks for adding all the tests

File: test/components/views/messages/EncryptionEvent-test.tsx (hunk 2 above)
Comment (weeman1337) [reply test/components/views/messages/EncryptionEvent-test.tsx:2]: Migrated :heavy_check_mark:

**Pull request merged**